backend/FwHeadless/Controllers/MediaFileController.cs

```diff
@@ -0,0 +1,305 @@
+        if (string.IsNullOrEmpty(filename))
+        {
+            var mimeType = httpContext.Request.Headers.ContentType.FirstOrDefault();
+            var ext = MimeUtility.GetExtensions(mimeType ?? "")?.FirstOrDefault() ?? "bin";
+            filename = $"file.{ext}";
+        }
+        if (fileId is null || fileId.Value == default)
+        {
+            fileId = Guid.NewGuid();
+        }
+        if (metadata is null)
+        {
+            metadata = JsonSerializer.Deserialize<FileMetadata>(metadataJson ?? "{}", JsonSerializerOptions.Web);
+            metadata ??= new FileMetadata();
+        }
+        else
+        {
+            var fromJson = JsonSerializer.Deserialize<FileMetadata>(metadataJson ?? "{}", JsonSerializerOptions.Web);
+            // If form fields specified, they should override any JSON passed in "metadata" field
+            if (fromJson is not null)
+            {
+                fromJson.Merge(metadata);
+                metadata = fromJson;
+            }
+        }
+        var mediaFile = await lexBoxDb.Files.FindAsync(fileId);
+        if (mediaFile is null && !newFilesAllowed)
+        {
+            return TypedResults.NotFound(); // PUT requests must modify an existing file and return 404 if it doesn't exist
+        }
+        if (mediaFile is null)
+        {
+            if (projectId is null)
+            {
+                return TypedResults.BadRequest(FileUploadErrorMessage.ProjectIdRequiredForNewFiles);
+            }
+            // If no filename specified in form, get it from uploaded file
+            if (string.IsNullOrEmpty(filename)) filename = file.FileName;
+            mediaFile = new MediaFile()
+            {
+                Id = fileId.Value,
+                Filename = Path.Join(fileId.Value.ToString(), filename),
+                ProjectId = projectId.Value,
+                Metadata = metadata,
+            };
+            lexBoxDb.Files.Add(mediaFile);
+        }
+        else
+        {
+            replacedExistingFile = true;
+            projectId ??= mediaFile.ProjectId;
+            if (projectId != mediaFile.ProjectId)
+            {
+                return TypedResults.BadRequest(FileUploadErrorMessage.UploadedFilesCannotBeMovedToNewProjects);
+            }
+            filename = Path.GetFileName(mediaFile.Filename); // Strip GUID prefix for consistency with create-file path; we'll be adding it back later
+            if (mediaFile.Metadata is null) mediaFile.Metadata = metadata ?? new FileMetadata();
+            else mediaFile.Metadata.Merge(metadata ?? new FileMetadata());
+        }
+        var project = await lexBoxDb.Projects.FindAsync(projectId);
+        if (project is null) return TypedResults.NotFound();
+        var projectFolder = config.Value.GetProjectFolder(project.Code, projectId.Value);
+        if (projectFolder is null)
```

Comment:
the error indicates that the folder doesn't exist, but this is just checking if the variable is null. Should probably actually check if the folder exists

Reply:
Done.